### ISSUE TITLE
Rotation/translation API questions

### ISSUE THREAD

**Issue by clarkbab** [eigenvivek/DiffDRR#409]
Hi there, thanks for developing this tool!

I'm just a little confused by the rotation/translation values in [the introduction](https://github.com/eigenvivek/DiffDRR/blob/main/notebooks/tutorials/introduction.ipynb). Specifically this part in "Rendering multiple DRRs at once":

```python
rot = torch.tensor([[0.0, 0.0, 0.0], [0.0, 0.0, torch.pi]], device=device)
xyz = torch.tensor([[0.0, 850.0, 0.0], [0.0, 850.0, 0.0]], device=device)
img = drr(rot, xyz, parameterization="euler_angles", convention="ZXY")
```

<img width="810" height="381" alt="Image" src="https://github.com/user-attachments/assets/7a324a58-a777-4c3a-937e-0175b239fb50" />

1. From the commands it looks like we'd expect a 180 degree flip around the y-axis (rot=[0.0, 0.0, torch.pi], convention="ZXY"). However, the image has been flipped around LR (x-axis)? I thought maybe this is defined in camera coords (z=principal axis, x/y aligned with detector), but that would also be a flip around the x-axis.
2. With the translation param, I expected this to be in camera coords (instead of world coords where translations would need to be rotated also), and the underlying code looks like this is the case with the extrinsic matrix being calculated [R | Rt]. But shouldn't we be translating along the z-axis (camera's principle axis) then?
3. Does convention="ZXY" apply to the rotation only?

I guess I'm just trying to understand what these x/y/z translations/rotations represent and in which coordinate systems they're expressed? Forgive me if this is already documented :)

Thanks,
Brett

**Comment by clarkbab** [comment]
Thanks for the quick reply! I'll check out nanodrr, it might be exactly what I'm after!

**Comment by clarkbab** [comment]
Sorry, I haven't had a chance to migrate to nanodrr as all the code I've written is for DiffDRR. 

I had a further question: is it possible to set the centre of rotation to something other than the image centre? I'm generating DRRs from a planning CT and trying to replicate the positions of x-ray images taken during treatment. The treatment isocentre is known and different from the CT isocentre. I'm adjusting the CT affine so that the isocentre is at the treatment isocentre, but DiffDRR seems to be just setting the isocentre to the voxel centre of the image for every affine passed. "DRR subject canon." shows the affine within diffdrr after the "canonicalize" call.

Is it intentional that the centre of rotation is always the voxel centre of the image? Or is there some way to set it to another point?

Thanks,
Brett

When adjusting affine so treatment iso is at world iso:
```python
volume = tio.ScalarImage(tensor=volume, affine=affine)
subject = read(volume)
>
DRR volume: [[   1.171875      0.            0.         -296.0122118 ]
 [   0.            1.171875      0.         -300.46636909]
 [   0.            0.            1.         -168.01905823]
 [   0.            0.            0.            1.        ]]
DRR density: [[   1.171875      0.            0.         -296.0122118 ]
 [   0.            1.171875      0.         -300.46636909]
 [   0.            0.            1.         -168.01905823]
 [   0.            0.            0.            1.        ]]
DRR subject: [[   1.171875      0.            0.         -296.0122118 ]
 [   0.            1.171875      0.         -300.46636909]
 [   0.            0.            1.         -168.01905823]
 [   0.            0.            0.            1.        ]]
DRR subject canon. [[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -299.4140625]
 [   0.           0.           1.        -175.       ]
 [   0.           0.           0.           1.       ]]
```

Without adjustment:
```python
volume = tio.ScalarImage(tensor=volume, affine=affine)
subject = read(volume)
>
[[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -483.9140625]
 [   0.           0.           1.        -505.       ]
 [   0.           0.           0.           1.       ]]
DRR volume: [[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -483.9140625]
 [   0.           0.           1.        -505.       ]
 [   0.           0.           0.           1.       ]]
DRR density: [[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -483.9140625]
 [   0.           0.           1.        -505.       ]
 [   0.           0.           0.           1.       ]]
DRR subject: [[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -483.9140625]
 [   0.           0.           1.        -505.       ]
 [   0.           0.           0.           1.       ]]
DRR subject canon. [[   1.171875     0.           0.        -299.4140625]
 [   0.           1.171875     0.        -299.4140625]
 [   0.           0.           1.        -175.       ]
 [   0.           0.           0.           1.       ]]
```

**Comment by clarkbab** [comment]
Update: I found the "center_volume" param and set it to False :) Thanks!

**Comment by clarkbab** [comment]
Oh I see, but the single-channel approach doesn't handle overlapping labels - e.g. overlapping GTV and lung masks. Any plans to convert to multi-channel in future?

**Comment by eigenvivek** [comment]
@clarkbab closing for now, feel free to open a new issue if you have any more questions
